### PR TITLE
chore: add SSI Credential Issuer

### PR DESCRIPTION
**Pull request by mgarciaLKS**
## Description

This pull request adds the SSI Credential Issuer to the umbrella chart. 
It was proposed by my co-worker @gomezbc in the 29/07 meeting and has been carried out by @gomezbc and myself according to his approval.

- In which Shared Services file should I enable the SSI Credential Issuer for the tests?

## Pre-review checks

Please ensure to do as many of the following checks as possible, before asking for committer review:

- [x] DEPENDENCIES are up-to-date. [Dash license tool](https://github.com/eclipse/dash-licenses). Committers can open IP issues for restricted libs.
- [x] [Copyright and license header](https://eclipse-tractusx.github.io/docs/release/trg-7/trg-7-02) are present on all affected files

### CODE REVIEW

File: charts/umbrella/README.md [hunk 1]
```diff
@@ -117,6 +117,7 @@ If you still face DNS issues, add the hosts to your /etc/hosts file:
 192.168.49.2    portal-backend.tx.test
 192.168.49.2    semantics.tx.test
 192.168.49.2    sdfactory.tx.test
+192.168.49.2    ssi.tx.test
```
Comment (evegufy) [inline charts/umbrella/README.md:120]: ssi-credential-issuer.tx.test

File: charts/umbrella/README.md [hunk 2]
```diff
@@ -144,6 +145,7 @@ For Windows edit the hosts file under `C:\Windows\System32\drivers\etc\hosts`:
...
```
Comment (evegufy) [inline charts/umbrella/README.md:148]: ssi-credential-issuer.tx.test

File: charts/umbrella/README.md [hunk 3]
```diff
@@ -426,6 +429,7 @@ Currently enabled ingresses:
   - http://portal-backend.tx.test/api/notification/swagger/index.html
 - http://portal.tx.test
 - http://semantics.tx.test/discoveryfinder/swagger-ui/index.html
+- http://ssi.tx.test/
```
Comment (evegufy) [inline charts/umbrella/README.md:432]: ssi-credential-issuer.tx.test

File: charts/umbrella/values.yaml [hunk 4]
```diff
@@ -46,6 +46,7 @@ portal:
   sdfactoryAddress: "http://sdfactory.tx.test"
   clearinghouseAddress: "http://validation.tx.test"
   clearinghouseTokenAddress: "http://keycloak.tx.test/realms/example/protocol/openid-connect/token"
+  issuerComponentAddress: "http://ssi.tx.test"
```
Comment (evegufy) [inline charts/umbrella/values.yaml:49]: ssi-credential-issuer.tx.test

File: charts/umbrella/values.yaml [hunk 5]
```diff
@@ -639,6 +640,125 @@ iatpmock:
           - path: /
             pathType: ImplementationSpecific
 
+ssi-credential-issuer:
+  enabled: false
+  portalBackendAddress: "http://portal-backend.tx.test"
+  walletAddress: "http://managed-identity-wallets.tx.test"
```
Comment (evegufy) [inline charts/umbrella/values.yaml:646]: http://iatpmock.tx.test

File: charts/umbrella/values.yaml [hunk 6]
```diff
@@ -639,6 +640,125 @@ iatpmock:
...
+  walletTokenAddress: "http://centralidp.tx.test/auth/realms/CX-Central/protocol/openid-connect/token"
```
Comment (evegufy) [inline charts/umbrella/values.yaml:647]: https://iatpmock-token-address.tx.test/oauth/token

File: charts/umbrella/values.yaml [hunk 7]
```diff
@@ -639,6 +640,125 @@ iatpmock:
...
+      # -- Client-secret for portal client-id. Secret-key 'portal-client-secret'.
+      clientSecret: "VRHQM2NOA4176Vrscxey1DdO4P7ikUQc"
+    logging:
+      default: "Debug"
+    processIdentity:
+      identityId: ac1cf001-7fbc-1f2f-817f-bce058020006
+    wallet:
+      # -- Provide wallet client-id from CX IAM centralidp.
+      # You must specify the technical user with the required roles for the interaction with the managed-identity-wallet
+      clientId: "sa-cl5-custodian-2"
```
Comment (evegufy) [inline charts/umbrella/values.yaml:684]: please change to the default of clientId: "wallet-client-id"

File: charts/umbrella/values.yaml [hunk 8]
```diff
@@ -639,6 +640,125 @@ iatpmock:
...
+      # -- Client-secret for wallet client-id. Secret-key 'wallet-client-secret'.
+      clientSecret: "UIqawwoohsvZ6AZOd1llLhnsUTKMWe4D"
```
Comment (evegufy) [inline charts/umbrella/values.yaml:686]: please leave that one empty for now "". as that's the clientid and secret for the iatp/dim wallet not the managed identity wallet

**Review by evegufy — CHANGES_REQUESTED** [review]

File: charts/umbrella/values.yaml [hunk 9]
```diff
@@ -639,6 +640,125 @@ iatpmock:
...
+
+  postgresql:
+    enabled: true
+    architecture: standalone
+    primary:
+      persistence:
+        enabled: true
+    auth:
+      # -- Non-root username.
+      username: issuer
```
Comment (evegufy) [inline charts/umbrella/values.yaml:716]: I think this value is the same as the default value, see https://github.com/eclipse-tractusx/ssi-credential-issuer/blob/v1.0.0/charts/ssi-credential-issuer/values.yaml, if so, please remove

File: charts/umbrella/values.yaml [hunk 10]
```diff
@@ -639,6 +640,125 @@ iatpmock:
...
+      # -- Database name.
+      database: issuer
```
Comment (evegufy) [inline charts/umbrella/values.yaml:718]: I think this value is the same as the default value, see https://github.com/eclipse-tractusx/ssi-credential-issuer/blob/v1.0.0/charts/ssi-credential-issuer/values.yaml, if so, please remove

File: charts/umbrella/values.yaml [hunk 11]
```diff
@@ -639,6 +640,125 @@ iatpmock:
...
+      # -- Secret containing the passwords for root usernames postgres and non-root username issuer.
+      # Should not be changed without changing the "issuer-postgresSecretName" template as well.
+      existingSecret: "{{ .Release.Name }}-issuer-postgres"
```
Comment (evegufy) [inline charts/umbrella/values.yaml:721]: I think this value is the same as the default value, see https://github.com/eclipse-tractusx/ssi-credential-issuer/blob/v1.0.0/charts/ssi-credential-issuer/values.yaml, if so, please remove

File: charts/umbrella/values.yaml [hunk 12]
```diff
@@ -639,6 +640,125 @@ iatpmock:
...
+          encryptionKey: "deb8261ec7b89c344f1c5ef5a11606e305f14e0d231b1357d90ad0180c5081d3"
+    
+  issuermigrations:
+    logging:
+      default: "Debug"
+
+  credentialExpiry:
+    expiry:
+      expiredVcsToDeleteInMonth: 12
+      inactiveVcsToDeleteInWeeks: 12
```
Comment (evegufy) [inline charts/umbrella/values.yaml:700]: I think this values are the same as the default values, see https://github.com/eclipse-tractusx/ssi-credential-issuer/blob/v1.0.0/charts/ssi-credential-issuer/values.yaml, if so, please remove

File: charts/umbrella/values.yaml [hunk 13]
```diff
@@ -639,6 +640,125 @@ iatpmock:
...
+    logging:
+      default: "Debug"
+
+  dbConnection:
```
Comment (evegufy) [inline charts/umbrella/values.yaml:704]: I think this values are the same as the default values, see https://github.com/eclipse-tractusx/ssi-credential-issuer/blob/v1.0.0/charts/ssi-credential-issuer/values.yaml, if so, please remove

**Review by evegufy — CHANGES_REQUESTED** [review]

File: charts/umbrella/values.yaml [hunk 14]
```diff
@@ -639,6 +640,125 @@ iatpmock:
...
+
+  issuer:
+    swaggerEnabled: true
+    logging:
+      businessLogic: "Debug"
+      default: "Debug"
+    portal:
+      # -- Provide portal client-id from CX IAM centralidp.
+      # You must specify the technical user with the required roles for the interaction with the portal
+      clientId: "sa-cl24-01"
```
Comment (evegufy) [inline charts/umbrella/values.yaml:657]: I just checked again, and that's the correct clientId 👍 (sorry for the confusion in the call earlier)

File: charts/umbrella/values.yaml [hunk 15]
```diff
@@ -639,6 +640,125 @@ iatpmock:
...
+      encryptionConfigs:
+        index0:
+          # EncryptionKey for wallet. Secret-key 'credential-encryption-key0'. Expected format is 256 bit (64 digits) hex.
+          encryptionKey: "deb8261ec7b89c344f1c5ef5a11606e305f14e0d231b1357d90ad0180c5081d3"
+
+  processesworker:
+    portal:
+      # -- Provide portal client-id from CX IAM centralidp.
+      # You must specify the technical user with the required roles for the interaction with the portal
+      clientId: "sa-cl24-01"
```
Comment (evegufy) [inline charts/umbrella/values.yaml:674]: I just checked again, and that's the correct clientId 👍 (sorry for the confusion in the call earlier)

**Comment by evegufy** [comment]
> #### ⚠️ GitGuardian has uncovered 3 secrets following the scan of your pull request.
> Please consider investigating the findings and remediating the incidents. Failure to do so may lead to compromising the associated services or software components.
> 
> Since your pull request originates from a forked repository, GitGuardian is not able to associate the secrets uncovered with secret incidents on your GitGuardian dashboard. Skipping this check run and merging your pull request will create secret incidents on your GitGuardian dashboard.
> 
> 🔎 Detected hardcoded secrets in your pull request
> 🛠 Guidelines to remediate hardcoded secrets
> 🦉 [GitGuardian](https://dashboard.gitguardian.com/auth/login/?utm_medium=checkruns&utm_source=github&utm_campaign=cr1) detects secrets in your source code to help developers and security teams secure the modern development process. You are seeing this because you or someone else with access to this repository has authorized GitGuardian to scan your pull request.

Hi @mgarciaLKS as those are test secrets, I skipped GitGuardian

**Comment by mgarciaLKS** [comment]
@evegufy Thanks for the feedback.

I have made the changes mentioned above, I hope they are correct this time.

On the other hand, would there be a way to skip GitGuardian automatically and also run the remaining tests? So I wouldn't have to wait for you to do it manually, and I can fix the tests that fail dynamically? 

Greetings.

File: charts/umbrella/values.yaml (hunk 14 above)
Comment (mgarciaLKS) [reply charts/umbrella/values.yaml:657]: thanks for the review

**Review by evegufy — CHANGES_REQUESTED** [review]

**Comment by evegufy** [comment]
> @evegufy Thanks for the feedback.
> 
> I have made the changes mentioned above, I hope they are correct this time.
> 
> On the other hand, would there be a way to skip GitGuardian automatically and also run the remaining tests? So I wouldn't have to wait for you to do it manually, and I can fix the tests that fail dynamically?
> 
> Greetings.

Hi @mgarciaLKS thanks for implementing the changes!

No, it's not possible to skip GitGuardian automatically but now that you're a project contributor (https://github.com/eclipse-tractusx/sig-infra/issues/539) the remaining test should be triggered automatically without manual approval.

File: charts/umbrella/values.yaml [hunk 16]
```diff
@@ -639,6 +640,125 @@ iatpmock:
...
+      # You must specify the technical user with the required roles for the interaction with the managed-identity-wallet
+      clientId: "sa-cl5-custodian-2"
+      # -- Client-secret for wallet client-id. Secret-key 'wallet-client-secret'.
+      clientSecret: "UIqawwoohsvZ6AZOd1llLhnsUTKMWe4D"
+      encryptionConfigIndex: 0
+      encryptionConfigs:
+        index0:
+          # EncryptionKey for wallet. Secret-key 'process-wallet-encryption-key0'. Expected format is 256 bit (64 digits) hex.
+          encryptionKey: "deb8261ec7b89c344f1c5ef5a11606e305f14e0d231b1357d90ad0180c5081d3"
+
```
[the remark this answers is not in the document]
Comment (mgarciaLKS) [reply charts/umbrella/values.yaml]: Done :)

**Comment by evegufy** [comment]
> @evegufy One question, now that the test are divided in two shared services to optimize resources, in which Shared Services file should I enable the SSI Credential Issuer for the tests?

Hi @mgarciaLKS right, forgot about that, could you please add it to file 2?

**Comment by mgarciaLKS** [comment]
@evegufy It seems to have worked :)

**Review by evegufy — APPROVED** [review]

**Pull request merged**